Documentation/exposing-prometheus-and-alertmanager.md

```diff
@@ -0,0 +1,284 @@
+    targetPort: web
+  selector:
+    alertmanager: main
+```
+
+And the Alertmanager web UI will be available at `http://node-ip:30903/`.
+
+## kubectl proxy
+
+The Prometheus and Alertmanager web UIs may only be interesting to cluster
+admins, or people who access the cluster with `kubectl` regularly. In this case 
+it seems natural to access those using `kubectl proxy`.
+
+To be able to do so, create a `Service` of type `ClusterIP`.
+
+```
+apiVersion: v1
+kind: Service
+metadata:
+  name: prometheus-main
+spec:
+  type: ClusterIP
+  ports:
+  - name: web
+    port: 9090
+    protocol: TCP
+    targetPort: web
+  selector:
+    prometheus: main
+```
+
+Once we now create the Prometheus object and all `Pod`s have been created
+properly, and we execute
+
+```
+kubectl proxy
+```
+
+the web UI will be accessible via
+http://127.0.0.1:8001/api/v1/proxy/namespaces/default/services/prometheus-main:web/
+
+The last caveat about this is that Prometheus and Alertmanager need to be
+configured with the full URL they are going to be exposed at. Therefore the
+`Prometheus` manifest will need an entry for `externalUrl`.
+
+```
+apiVersion: "monitoring.coreos.com/v1alpha1"
+kind: "Prometheus"
+metadata:
+  name: "main"
+spec:
+  replicas: 1
+  version: v1.5.0
+  externalUrl: http://127.0.0.1:8001/api/v1/proxy/namespaces/default/services/prometheus-main:web/
+  routePrefix: /
+  resources:
+    requests:
+      memory: 400Mi
+```
+
+Fortunately the Alertmanager is able to perform all requests relative to its
+root when using this approach so the manifest for the `Alertmanager` object is
+simply the following.
+
+```
+apiVersion: "monitoring.coreos.com/v1alpha1"
+kind: "Alertmanager"
+metadata:
+  name: "main"
+spec:
+  replicas: 3
+  version: v0.5.1
+  resources:
+    requests:
+      memory: 400Mi
+```
+
+And with the respective `Service`.
+
+```
+apiVersion: v1
+kind: Service
+metadata:
+  name: alertmanager-main
+spec:
+  type: ClusterIP
+  ports:
+  - name: web
+    port: 9093
+    protocol: TCP
+    targetPort: web
+  selector:
+    alertmanager: main
+```
+
+Then it will be available under
+http://127.0.0.1:8001/api/v1/proxy/namespaces/default/services/alertmanager-main:web/
+
+## Ingress
+
+Exposing the Prometheus or Alertmanager web UI via an `Ingress` object is
+requires a running ingress controller. If you are unfamiliar with Ingress, have
+a look at the [documentation](https://kubernetes.io/docs/user-guide/ingress/).
+
+This example was tested with the [nginx ingress
+controller](https://github.com/kubernetes/contrib/tree/master/ingress/controllers/nginx).
+For a quickstart for running the nginx ingress controller run:
+
+```
+kubectl create -f https://raw.githubusercontent.com/kubernetes/contrib/master/ingress/controllers/nginx/rc.yaml
+```
+
+> It is highly recommended to compare different ingress controllers for a
+> production environment. The nginx ingress controller may or may not be what
+> is suitable for your production environment.
```

Comment:
Maybe also suggest alternatives, like "traefik".
It's a common assumption that the nginx ingress is the only implementation available.